### PR TITLE
[Form] Added CountryType option for using alpha3 country codes

### LINKED ISSUE #20313: [Intl][Form] Support ISO 3166-1 Alpha-3 country codes
Could we please extend the form type `CountryType` via an option to make it support ISO_3166-1 Alpha-3 country codes?

Sources for the country codes:
- https://www.iso.org/obp/ui/
- http://en.wikipedia.org/wiki/ISO_3166-1_alpha-3
- https://github.com/thephpleague/iso3166
- https://gist.github.com/brianherbert/1687655

I'm currently working on a project where I need country codes as 3 characters instead of just 2.

Would it be possible to include a new alpha3 option or something to support 3 characters country codes? 

(This was first raised in #8492 but that issue was closed without being resolved properly)

### PR DESCRIPTION
**Pull request by creiner**
| Q             | A
| ------------- | ---
| Branch?       | 4.4 
| Bug fix?      | no
| New feature?  | yes
| Deprecations? | no 
| Tickets       | Fixes #20313
| License       | MIT
| Doc PR        | -

In the linked issue #20313 was a proposal to add an alpha3 option to the country type.
Here it is..
Hopefully I've made no mistake, so when the code is fine, I will create a documentation PR.. :-)

### CODE REVIEW

File: src/Symfony/Component/Form/Extension/Core/Type/CountryType.php [hunk 1]
```diff
@@ -42,13 +42,15 @@ public function configureOptions(OptionsResolver $resolver)
...
 
-                return new IntlCallbackChoiceLoader(function () use ($choiceTranslationLocale) {
-                    return array_flip(Countries::getNames($choiceTranslationLocale));
+                return new IntlCallbackChoiceLoader(function () use ($choiceTranslationLocale, $alpha3) {
+                    return array_flip($alpha3 ? Countries::getAlpha3Names($choiceTranslationLocale) : Countries::getNames($choiceTranslationLocale));
                 });
             },
             'choice_translation_domain' => false,
             'choice_translation_locale' => null,
+            'alpha3' => false,
```
Comment (ro0NL) [inline src/Symfony/Component/Form/Extension/Core/Type/CountryType.php]: you should set the allowed types (bool) below as well

File: src/Symfony/Component/Form/Tests/Extension/Core/Type/CountryTypeTest.php [hunk 2]
```diff
@@ -58,6 +58,25 @@ public function testChoiceTranslationLocaleOption()
         $this->assertContainsEquals(new ChoiceView('MY', 'MY', 'Малайзія'), $choices);
     }
 
+    /**
+     * @requires extension intl
```
Comment (ro0NL) [inline src/Symfony/Component/Form/Tests/Extension/Core/Type/CountryTypeTest.php]: i believe this is only needed with `choice_translation_locale` which triggers a different collation for sorting

File: src/Symfony/Component/Form/Tests/Extension/Core/Type/CountryTypeTest.php [hunk 3]
```diff
@@ -58,6 +58,25 @@ public function testChoiceTranslationLocaleOption()
...
+     */
+    public function testAlpha3Option()
```
Comment (ro0NL) [inline src/Symfony/Component/Form/Tests/Extension/Core/Type/CountryTypeTest.php]: perhaps we should add a 2nd test (testAlpha3OptionWithTranslationLocale) to ensure alpha3 codes are translatable as well. See inspiration above.

**Review by ro0NL — COMMENTED** [review]
the same applies to LanguageType, isnt it?

File: src/Symfony/Component/Form/Tests/Extension/Core/Type/CountryTypeTest.php (hunk 3 above)
Comment (creiner) [reply src/Symfony/Component/Form/Tests/Extension/Core/Type/CountryTypeTest.php]: I've added it!

File: src/Symfony/Component/Form/Tests/Extension/Core/Type/CountryTypeTest.php (hunk 2 above)
Comment (creiner) [reply src/Symfony/Component/Form/Tests/Extension/Core/Type/CountryTypeTest.php]: Aaah good to know.. I've removed it..

File: src/Symfony/Component/Form/Extension/Core/Type/CountryType.php (hunk 1 above)
Comment (creiner) [reply src/Symfony/Component/Form/Extension/Core/Type/CountryType.php]: Good idea.. i've added it..

**Comment by creiner** [comment]
I've added it to the LanguageType and added also the tests...

When a language doesn't have a three letter code, it doesn't appear in the collection (behavior of Language::getAlpha3Names ).. 
Maybe this can cause confusion.. do you think a fallback to 2-letter codes for languages which doesn't have three letter codes would be appropriate?

**Review by ro0NL — COMMENTED** [review]
LGTM. I think you need to bump symfony/intl in composer.json of the Form component still ... see https://travis-ci.org/symfony/symfony/jobs/592410645#L4077

> do you think a fallback to 2-letter codes for languages which doesn't have three letter codes would be appropriate?

i'd like to validate the dataset further in #33165, feel free to share any data related issues there. In general the alpha2 and alpha3 list should be the same size. For languages the alpha2 lists also includes alpha3 codes in case no alpha2 code is available, whereas Countries should have alpha3 codes for each alpha2 code.

**Review by ro0NL — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @creiner.

**Pull request merged**